sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -68,6 +68,11 @@ def _get_list_element_type(typ: Optional[type]) -> Optional[type]:
     if typ is None:
         return None
 
+    # Annotations not specifying the element type are assumed by mypy
+    # to signify Any element type. Follow suit here.
+    if typ in {list, List, Sequence, abc.Sequence}:
+        return cast(type, Any)
```

Comment:
I'm good with the way you have it so feel free to ignore...

Just wondering if we could put this after the origin check so we don't have to have the `typ in {...}` twice? Granted I have not tried it out.

```python
def _get_list_element_type(typ: Optional[type]) -> Optional[type]:
    if typ is None:
        return None

    # If typ is a list, get the type for its values, to pass
    # along for each item.
    origin = _types.get_origin(typ)
    if typ is list or origin in {list, List, Sequence, abc.Sequence}:
        args = _types.get_args(typ)
        # Return the element type, if available, otherwise Any to be consistent with
        # how mypy treats annotations that do not specify an element type.
        return args[0] if args else cast(type, Any)

    raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
```

Reply:
This doesn't quite work since `_types.get_origin(Sequence) is None`